packages/amplify_authenticator/lib/src/blocs/auth/auth_bloc.dart

```diff
@@ -143,6 +143,38 @@ class StateMachineBloc {
     }
   }
 
+  Stream<AuthState> _isValidSession() async* {
+    try {
+      bool isValidSession = await _authService.isValidSession();
+      if (isValidSession) {
+        yield const Authenticated();
+      } else {
+        /// [isValidSession] returns false if the native platform
+        /// returns a [SignedOutException] or if the native libraries return a session object but
+        /// UserPool tokens are null (when unauthenticated access is enabled on Identity Pool).
+        yield AuthFlow.signin;
+      }
+    } on SessionExpiredException {
+      /// In this case, we want to give the end user an exception message and go to signin.
+      _exceptionController.add(const AuthenticatorException(
+          'Your session has expired. Please sign in.',
+          showBanner: true));
+      yield AuthFlow.signin;
+
+      /// On [AmplifyException] and [Exception], update exception controller, do not show banner
+      /// and go to sign in screen.
+    } on AmplifyException catch (e) {
+      _exceptionController
+          .add(AuthenticatorException(e.message, showBanner: false));
+      yield AuthFlow.signin;
+    } on Exception {
+      _exceptionController.add(const AuthenticatorException(
+          'An unknown exception has occurred when fetching the session',
```

Comment:
Probably just pass back the exception, no?

`AuthenticatorException(e.toString())`

Reply:
@dnys1 Makes sense - done.